auto_nag/topcrash.py

```diff
@@ -0,0 +1,97 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from datetime import datetime
+from typing import Dict, Optional, Union
+
+import libmozdata.socorro as socorro
+from libmozdata import utils as lmdutils
+
+
+class Topcrash:
+    def __init__(
+        self,
+        minimum_crashes: Optional[int] = 0,
+        minimum_startup_crashes: Optional[int] = 0,
+    ) -> None:
+        """Constructor
+
+        Args:
+            minimum_crashes: the minimum number of crashes to consider a
+                signature in the top crashes.
+            minimum_startup_crashes: the minimum number of crashes to consider a
+                signature as a startup crash.
+        """
+        self.minimum_crashes = minimum_crashes
+        self.minimum_startup_crashes = minimum_startup_crashes
+
+    def get_signatures(
+        self,
+        date: Union[str, datetime],
+        duration: Optional[int] = 11,
+        tc_limit: Optional[int] = 50,
+    ) -> Dict[str, bool]:
+        """Fetch the top crashes from socorro.
+
+        Top crashes will be queried twice for each release channel, one that
+        targets startup crashes and another that targets all crashes.
+
+        Args:
+            date: the final date
+            duration: the number of days to retrieve the crash data
+            tc_limit: the number of topcrashes to load on each query
+
+        Returns:
+            A dictionary where the keys are crash signatures, and the values are
+            booleans that indicate whether a crash is a startup crash or not.
+        """
+
+        start_date = lmdutils.get_date(date, duration - 1)
+        end_date = lmdutils.get_date(date)
+        date_range = socorro.SuperSearch.get_search_date(start_date, end_date)
+
+        params_combinations = [
+            {
+                "product": "Firefox",
```

Comment:
I wonder if we should support Fenix too.

Reply:
@marco-c wdyt?

Reply:
We can ask on Slack in the #mobile channel

Reply:
Fenix team is onboard. Now we need the [crash identification criteria on the wiki page](https://wiki.mozilla.org/CrashKill/Topcrash) to be updated for Fenix.

Reply:
Thanks to @gabrielesvelto, he updated the wiki page. 

The criteria for Fenix added in 9392dcad7fb5f991e6bc7d79ce34e3dc4ca9580b.